docs/Lua API.rst

```diff
@@ -4079,10 +4079,22 @@ DFHack core invokes the scripts in the *core context* (see above);
 however it is possible to call them from any lua code (including
 from other scripts) in any context, via the same function the core uses:
 
-* ``dfhack.run_script(name[,args...])``
-
+* ``dfhack.run_script(command[, ...])``
+   
   Run a lua script in hack/scripts/, as if it was started from dfhack command-line.
-  The ``name`` argument should be the name stem, as would be used on the command line.
+  The syntax for the argument is similar to dfhack.run_command.
+  The command can be passed as a table, multiple string
+  arguments, or a single string argument (not recommended - in this case, the
+  usual DFHack console tokenization is used).
+  The following examples are equivalent::
+        dfhack.run_script({'ls', '-a'})
+        dfhack.run_script('ls', '-a')
+        dfhack.run_script('ls -a')  -- not recommended
+  
+  Unlike ``dfhack.run_command``, ``dfhack.run_script`` is capable of taking the 
+  return value of a lua script as an argument. 
+  For example::
+        dfhack.run_scipt("process", dfhack.gui.getDwarfmodeViewDims());
```

Comment:
Also to clarify @warmist's comment: while this is possible, it isn't expected to work, so I don't think we want to include it here. I believe passing non-strings to `run_command()` throws an error immediately, and if we were to reimplement `run_script()` today, it probably would as well.

Reply:
Not to be a contrarian, but passing commands to run_script actually does work without an error. 
Example:

test.lua:
`dfhack.run_script("personal/print", dfhack.gui.getDwarfmodeViewDims());
`
personal/print.lua:
```
local args = {...}
print(args)
```

Return from print.lua:
`table: 00000202BD97FAE0`

Although, I really don't know what use this has, and proving its existence is really little more than a proof of concept.